src/main/webapp/sw.js

```diff
@@ -0,0 +1,82 @@
+// Service Worker
+
+const CACHE_NAME = "codeu-chat"  // codeu-chat is arbitrary name of chat 
+
+/**
+ * Caches resources that don't change often (ex. CSS, images),
+ * and resources needed for the service worker (ex. error pages, alternate offline versions)
+ */
+async function installDeps() {
+    const cache = await caches.open(CACHE_NAME); 
+    return cache.addAll([
+        "/css/main.css",
+
+        "/offline.html", // General error page
+        "/?offline", // Homepage
+        "/about.jsp?offline", // About
+    ]);
+}
+
+/**
+ * Called to load a response from the network or cache.
+ * @param {Request} request from user
+ * @returns response from server or cache
+ */
+async function load(request) {
+    let response;
+    try {
+        response = await fetch(request);
+    } catch (err) {
+        // If no internet, return from cache instead
+        const response = await caches.match(request); 
+        // if no matching cache item, return a special response
+        if (response == null) return findOfflinePage(request);
```

Comment:
What error would you get such that findOfflinePage should be executed here?

Reply:
I set brakepoint at that line and when I go to about page in offline mode, i get "TypeError: Failed to fetch"